store/src/java/com/zimbra/cs/event/analytics/contact/ContactAnalytics.java

```diff
@@ -0,0 +1,112 @@
+package com.zimbra.cs.event.analytics.contact;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.event.EventStore;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+import java.time.temporal.WeekFields;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public class ContactAnalytics {
+    public enum TimeRange {
+        LAST_DAY("NOW-1DAY", "NOW", "+1HOUR", "HOUR"),
```

Comment:
I'm not sure that the enum constructor is the best place to specify these parameters. They are essentially a solr implementation detail, which the `ContactAnalytics` package should ideally be agnostic of. I recommend moving them to `SolrEventStore`, perhaps a static map or a method with a switch statement?